docker-compose.yml

```diff
@@ -24,7 +24,7 @@ services:
       - /etc/ssl/dhparams.pem:/etc/ssl/dhparam.pem
       - /var/www/chameleon/static:/var/www/chameleoncloud.org/static/
       - ./conf/nginx/nginx.conf:/etc/nginx/nginx.conf
-      - ./conf/nginx/api.chameleoncloud.org.conf:/etc/nginx/conf.d/api.chameleoncloud.server.conf
+      - ./conf/nginx/conf.d:/etc/nginx/conf.d
```

Comment:
This would work here for now, but if we ever have any dev specific server or include configs they could break production unexpectedly (certs/keys may not resolve for example), we should use  environment specific includes in the compose file and check in the includes file that does the redirect to allocations api so the nginx config is complete

Reply:
In my opinion environment-specific configs should be handled at a layer above camino, dropping/templating the relevant configs into this location. What if we want to add HTTP authentication to a portal endpoint and put the simple password into the nginx conf. We can't check that into the repo, so then where does it live? Such issues are why we use config management and tooling in the chi-in-a-box repo.

How were you thinking we can test the reference API on dev, as another example of this problem? There is no dev config for the reference API host checked in that I could see.

My preference would actually be to make conf.d an empty directory and not under source control, and then generate the reference API and allocations API configurations externally and place them there as part of the portal deployment. As a half-measure I'll add the allocations conf to this PR so at least we don't have a broken current state w.r.t configuration and source control.

Hope that makes sense

Reply:
Thanks, I think checking in the includes config goes a long way towards al sure we don't lose the changes. As far as reference api, we talk about what type of testing you're thinking of, but there is a separate dev config to point to different reference-api containers in dev. Let's chat some more about specifics, this looks good for now, thanks!